cmake/CMakeLists.txt

```diff
@@ -1003,32 +1035,108 @@ if (CPUINFO_SUPPORTED)
     string(APPEND CMAKE_CXX_FLAGS " -DCPUINFO_SUPPORTED")
   endif()
 
-  if (onnxruntime_PREFER_SYSTEM_LIB)
-    find_package(cpuinfo)
-  endif()
-  if (cpuinfo_FOUND)
-    message("Use cpuinfo from preinstalled system lib")
-  else()
-    message("Use cpuinfo from submodule")
-    set(CPUINFO_BUILD_TOOLS OFF CACHE INTERNAL "")
-    set(CPUINFO_BUILD_UNIT_TESTS OFF CACHE INTERNAL "")
-    set(CPUINFO_BUILD_MOCK_TESTS OFF CACHE INTERNAL "")
-    set(CPUINFO_BUILD_BENCHMARKS OFF CACHE INTERNAL "")
-    add_subdirectory(external/pytorch_cpuinfo EXCLUDE_FROM_ALL)
-    if (MSVC)
-      target_compile_options(cpuinfo PRIVATE "-D_CRT_SECURE_NO_WARNINGS")
-    endif()
-  endif()
+
+  set(CPUINFO_BUILD_TOOLS OFF CACHE INTERNAL "")
+  set(CPUINFO_BUILD_UNIT_TESTS OFF CACHE INTERNAL "")
+  set(CPUINFO_BUILD_MOCK_TESTS OFF CACHE INTERNAL "")
+  set(CPUINFO_BUILD_BENCHMARKS OFF CACHE INTERNAL "")
+    
+  FetchContent_Declare(
+    pytorch_cpuinfo
+    URL ${DEP_URL_pytorch_cpuinfo}
+	URL_HASH SHA1=${DEP_SHA1_pytorch_cpuinfo}
+    FIND_PACKAGE_ARGS NAMES cpuinfo
+  )
```

Comment:
Do we want to let XNNPAK know that it does not need to download cpu info and it this one?

Reply:
In XNNPack.cmake we have
```cmake
set(XNNPACK_USE_SYSTEM_LIBS ON CACHE INTERNAL "")
```

It is for the purpose.